packages/nx/internal-testing-utils/temp-fs.ts

```diff
@@ -10,19 +10,29 @@ import {
   rmSync,
   unlinkSync,
 } from 'fs-extra';
-import { joinPathFragments } from '../path';
+import { joinPathFragments } from '../src/utils/path';
 import { appendFileSync, existsSync, renameSync, writeFileSync } from 'fs';
+import { setWorkspaceRoot, workspaceRoot } from '../src/utils/workspace-root';
 
 type NestedFiles = {
   [fileName: string]: string;
 };
 
 export class TempFs {
   readonly tempDir: string;
+
+  private previousEnvWorkspaceRootPath: string;
+  private previousWorkspaceRoot: string;
+
   constructor(private dirname: string, overrideWorkspaceRoot = true) {
     this.tempDir = realpathSync(mkdtempSync(join(tmpdir(), this.dirname)));
+
+    this.previousEnvWorkspaceRootPath = process.env.NX_WORKSPACE_ROOT_PATH;
+    this.previousWorkspaceRoot = workspaceRoot;
+
     if (overrideWorkspaceRoot) {
       process.env.NX_WORKSPACE_ROOT_PATH = this.tempDir;
```

Comment:
Do we still need this? If we use `setWorkspaceRoot`?

Reply:
Yeah, I think thats fair. I'll update it to only use `setWorkspaceRoot`